src/zeta/aca_oam_server.cpp

```diff
@@ -0,0 +1,323 @@
+void ACA_Oam_Server::_parse_oam_flow_injection(uint32_t udp_dport, oam_message *oammsg)
+{
+  unsigned long not_care_culminative_time;
+  int overall_rc = EXIT_SUCCESS;
+
+  oam_match match = _get_oam_match_field(oammsg);
+
+  // check whether the udp_dport is the oam server port of the vpc
+  if (!_check_oam_server_port(udp_dport, match)) {
+    return;
+  }
+
+  oam_action action = _get_oam_action_field(oammsg);
+
+  string remote_host_ip = action.node_nw_dst;
+  string vpc_id = match.vni;
+  alcor::schema::NetworkType network_type = alcor::schema::NetworkType::VXLAN;
+
+  if (!aca_is_port_on_same_host(remote_host_ip)) {
+    ACA_LOG_INFO("%s", "port_neighbor not exist!\n");
+    string neighbor_id;
+    // get netigbor_id
+
+    //crate neighbor_port
+    aca_ovs_l2_programmer::ACA_OVS_L2_Programmer::get_instance().create_or_update_l2_neighbor(
+            neighbor_id, vpc_id, network_type, remote_host_ip,
+            (uint)stoi(vpc_id), not_care_culminative_time);
+  }
+  overall_rc = aca_oam_server::ACA_Oam_Server::_add_direct_path(match, action);
+
+  if (overall_rc == EXIT_SUCCESS) {
+    ACA_LOG_INFO("%s", "Command succeeded!\n");
+  } else {
+    ACA_LOG_ERROR("Command failed!!! overrall_rc: %d\n", overall_rc);
+  }
+
+  return;
+}
+
+void ACA_Oam_Server::_parse_oam_flow_deletion(uint32_t udp_dport, oam_message *oammsg)
+{
+  int overall_rc = EXIT_SUCCESS;
+  oam_match match = _get_oam_match_field(oammsg);
+  // check whether the udp_dport is the oam server port of the vpc
+  if (!_check_oam_server_port(udp_dport, match)) {
+    return;
+  }
+
+  overall_rc = _del_direct_path(match);
+
+  if (overall_rc == EXIT_SUCCESS) {
+    ACA_LOG_INFO("%s", "Command succeeded!\n");
+  } else {
+    ACA_LOG_ERROR("Command failed!!! overrall_rc: %d\n", overall_rc);
+  }
+
+  return;
+}
+
+void ACA_Oam_Server::_parse_oam_none(uint32_t /* in_port */, oam_message *oammsg)
+{
+  ACA_LOG_ERROR("Wrong OAM message type! (Message type = %d)\n", _get_message_type(oammsg));
+  return;
+}
+
+int ACA_Oam_Server::_add_direct_path(oam_match match, oam_action action)
+{
+  int overall_rc = EXIT_SUCCESS;
+
+  string vlan_id = to_string(aca_vlan_manager::ACA_Vlan_Manager::get_instance().get_or_create_vlan_id(
+          match.vni));
+  string outport_name =
+          aca_get_outport_name(alcor::schema::NetworkType::VXLAN, action.node_nw_dst);
+
+  string cmd_match = "ip,nw_proto=" + match.proto + ",nw_src=" + match.sip +
+                     ",nw_dst=" + match.dip + ",tp_src=" + match.sport +
+                     ",tp_dst=" + match.dport + ",dl_vlan=" + vlan_id;
+  string cmd_action = "action=\"strip_vlan,load:" + match.vni +
+                      "->NXM_NX_TUN_ID[],mod_dl_dst=" + action.inst_dl_dst +
+                      ",mod_nw_dst=" + action.inst_nw_dst +
+                      ",idle_timeout=" + action.idle_timeout + ",output:" + outport_name;
+
+  // Adding unicast rules in table20
+  string opt = "table=20,priority=50," + cmd_match + "," + cmd_action;
```

Comment:
great aproach, keep in mind table 20 (unicast) is not hooked up in current ACA yet, you will needed to hook it up :) using openflow rules.